### DISCUSSION TITLE
I have a .csv file and I want to bundle it in my apps, how can I do that by isar?

### DISCUSSION THREAD

**Discussion by HaloWang** [isar/isar#899] [Help]
I guess I should transform my .csv to specific format (like *.sqlite or *.db) and then add it to pubspec.yaml in my flutter project. But I can't find the related docs in your [website](https://isar.dev/tutorials/quickstart.html)

**Comment by cliftonlabrum** [comment]
I do something similar in one of my apps with a `.db` file. It would be the same process for a `.csv` or any other file type.

### Step 1
Put the file in `/assets/file.csv`

### Step 2
Register the asset in **pubspec.yaml**:
```
flutter:
  assets:
    - assets/file.csv
```

### Step 3
You can call a function from **main.dart** to load your file:
```
loadMyFile() async{
  ByteData data = await rootBundle.load('assets/file.csv');
  List<int> bytes = data.buffer.asUint8List(data.offsetInBytes, data.lengthInBytes);

  // Write and flush the bytes written
  final path = '...destination path of your file in the device's file system/file.csv'
  await File(path).writeAsBytes(bytes, flush: true);
}
```

### Step 4
You can now access the CSV file in your app, iterate over the lines of CSV, etc. to include the data in your Isar file.

Unless you need the data to be CSV, it may be easier to migrate the data to Isar on your own first, and then include the resulting `.isar` file as an asset and open it as another Isar instance when your app starts up.

Good luck!
